src/plot_api/plot_api.js

```diff
@@ -1580,6 +1580,12 @@ function _restyle(gd, aobj, _traces) {
                 var moduleAttrs = (contFull._module || {}).attributes || {};
                 var valObject = Lib.nestedProperty(moduleAttrs, ai).get() || {};
 
+                // if restyling entire attribute container, assume worse case
+                if(!valObject.valType) {
```

Comment:
It would because if it had a `valType` field it would only need a valid `role` (i.e. set to `'info' | 'style' | 'data'`) field to pass this [test](https://github.com/plotly/plotly.js/blob/caade64acd65f7a0dfa52932115a6aeee2aeb37a/test/jasmine/tests/plotschema_test.js#L38-L46).